propsigner/client.go

```diff
@@ -0,0 +1,66 @@
+package propsigner
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+
+	"github.com/filecoin-project/go-state-types/crypto"
+	"github.com/filecoin-project/specs-actors/actors/builtin/market"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+	pb "github.com/textileio/go-auctions-client/gen/wallet"
+)
+
+const (
+	maxResponseMessageSize = 100 << 10
+)
+
+// RequestSignatureV1 request a signature for a deal proposal to a remote wallet.
+func RequestSignatureV1(
+	ctx context.Context,
+	h host.Host,
+	authToken string,
+	proposal market.DealProposal,
+	remoteWallet peer.ID) (*crypto.Signature, error) {
```

Comment:
We basically ask for the auth-token, proposal to be signed, and remote wallet peer id; and we return the signature as a result.

There's a chance that this method parameters can change a bit after I use it in `dealerd`. Probably receiving a list of multiaddrs instead of the `peer.ID` to try every multiaddr as an option for connection. Nothing wild, but maybe tuned a bit (that's not entirely important anyway).